sir_cake/store/views.py

```diff
@@ -393,3 +398,9 @@ def _shipping_data_is_missing(request):
 def _delete_order_and_basket_items(order):
     BasketItem.objects.filter(order=order).delete()
     order.delete()
+
+
+def _get_basket_expire_date(preorder):
```

Comment:
You likely need to handle cases where `preorder` is `None` here because you send it from `_get_preorder_or_none`